pkg/trebuchet/server/apiserver/apiserver.go

```diff
@@ -0,0 +1,85 @@
+package apiserver
+
+import (
+	apis_trebuchet "github.com/atlassian/voyager/pkg/apis/trebuchet"
+	"github.com/atlassian/voyager/pkg/apis/trebuchet/install"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/version"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+)
+
+var (
+	Scheme = runtime.NewScheme()
+	Codecs = serializer.NewCodecFactory(Scheme)
+)
+
+func init() {
+	install.Install(Scheme)
+
+	metav1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
+
+	unversioned := schema.GroupVersion{Group: "", Version: "v1"}
+	Scheme.AddUnversionedTypes(unversioned,
+		&metav1.Status{},
+		&metav1.APIVersions{},
+		&metav1.APIGroupList{},
+		&metav1.APIGroup{},
+		&metav1.APIResourceList{},
+	)
+}
+
+type Config struct {
+	GenericConfig *genericapiserver.RecommendedConfig
+}
+
+// TrebuchetServer contains state for a Kubernetes cluster master/api server.
+type TrebuchetServer struct {
+	GenericAPIServer *genericapiserver.GenericAPIServer
+}
+
+type completedConfig struct {
+	GenericConfig genericapiserver.CompletedConfig
+}
+
+type CompletedConfig struct {
+	// Embed a private pointer that cannot be instantiated outside of this package.
+	*completedConfig
+}
+
+// Complete fills in any fields not set that are required to have valid data. It's mutating the receiver.
+func (cfg *Config) Complete() CompletedConfig {
+	c := completedConfig{
+		cfg.GenericConfig.Complete(),
+	}
+
+	c.GenericConfig.EnableDiscovery = false
+	c.GenericConfig.Version = &version.Info{
+		Major: "0",
+		Minor: "1",
+	}
+
+	return CompletedConfig{&c}
+}
+
+// New returns a new instance of TrebuchetServer from the given config.
+func (c completedConfig) New() (*TrebuchetServer, error) {
+	genericServer, err := c.GenericConfig.New("trebuchet-apiserver", genericapiserver.NewEmptyDelegate())
+	if err != nil {
+		return nil, err
+	}
+
+	s := &TrebuchetServer{
+		GenericAPIServer: genericServer,
+	}
+
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(apis_trebuchet.GroupName, Scheme, metav1.ParameterCodec, Codecs)
+
```

Comment:
leave a TODO comment here? This is where you will inject a custom Store or HTTP handler to proxy request to underlying service instead of storing in etcd like standard API server.